django/db/backends/mysql/validation.py

```diff
@@ -42,4 +43,16 @@ def check_field_type(self, field, field_type):
                     id='mysql.E001',
                 )
             )
+
+        if field.db_index and field_type.lower() in self.connection._limited_data_types:
+            errors.append(
+                checks.Warning(
+                    'MySQL does not support a database index on %s columns.'
+                    % field_type,
+                    hint="An index won't be created. Silence this warning if "
```

Comment:
I'd use this indent style: https://github.com/django/django/blob/98b3b14a648a128d409b8f6169c1a4ff9d71c1a4/tests/invalid_models_tests/test_ordinary_fields.py#L231-L234